tests/test_diag_coded_types.py

```diff
@@ -407,9 +412,10 @@ def test_encode_param_info_length_type_uint(self) -> None:
         odxlinks = OdxLinkDatabase()
         odxlinks.update({length_key_id: length_key})
         dct._resolve_odxlinks(odxlinks)
-        state = EncodeState(bytearray([0x10]), {length_key.short_name: 40})
-        byte_val = dct.convert_internal_to_bytes(0x12345, state, bit_position=0)
-        self.assertEqual(byte_val.hex(), "0000012345")
+        state = EncodeState(
+            coded_message=bytearray([0xcc]), parameter_values={}, cursor_byte_position=2)
+        dct.encode_into_pdu(0x12345, state)
+        self.assertEqual(state.coded_message.hex(), "cc00012345")
```

Comment:
why did the assertion changed here?

Reply:
it's been a while since I did this, so take this with a grain of salt: I suppose it is because this sets the cursor byte position to 2, and the initial value of the PDU to 0xcc. I don't remember why the old code produced two leading zero bytes...